### PR TITLE
dedup -- ignoreCase

### PR DESCRIPTION
**Pull request by raymondng**
Based on earlier review feedback from @leogrim 

cc: @andrewconner @2is10

### CODE REVIEW

**Comment by andrewconner** [comment]
That's a cool utility method.

**Comment by leogrim** [comment]
Indeed, but why not reuse the original method from RichContact?

**Comment by leogrim** [comment]
Actually, feel free to replace the one in RichContact with yours :)
:+1:

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala [hunk 1]
```diff
@@ -258,7 +258,7 @@ class TypeaheadCommander @Inject() (
         val kifi: Future[Seq[NetworkTypeAndHit]] = kifiF.map { hits => hits.map(hit => (SocialNetworks.FORTYTWO, hit)) }
         val abook: Future[Seq[NetworkTypeAndHit]] = abookF.map { hits =>
           val nonUsers = hits.filter(_.info.userId.isEmpty)
-          val distinctEmails = nonUsers.groupBy(_.info.email.address).map(_._2.head).toSeq.sortBy(_.score)
+          val distinctEmails = dedupBy(nonUsers) { _.info.email.address.toLowerCase }
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/TypeaheadCommander.scala]: Nice!

I think we normally use parentheses when passing a function that doesn’t name its argument, like on the previous line.

**Comment by raymondng** [comment]
Thanks for the feedback! Will revise based on input.

**Pull request merged**